### PR TITLE
Optional job_id parameter during query execution

### PR DESCRIPTION
**Pull request by fsnlla**
Context:
- to simplify integration with the `archive-query` service that relies on self-generated IDs, I propose to get an option to pass `job_id` as an optional parameter during query execution

In this PR:
- optional job_id support for caller-generated IDs
- update `get_job_status` function to support emptiness
- clippy suggestion

### CODE REVIEW

**Comment by thinkharderdev** [comment]
@fsnlla Not sure I understand the context here. Why are we passing a job ID? This is randomly generated by the scheduler when planning the query.

File: ballista/rust/scheduler/src/scheduler_server/grpc.rs [hunk 1]
```diff
@@ -406,7 +409,10 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> SchedulerGrpc
 
             debug!("Received plan for execution: {:?}", plan);
 
-            let job_id = self.state.task_manager.generate_job_id();
+            let job_id = match optional_job_id {
+                Some(OptionalJobId::JobId(job_id)) => job_id,
```
Comment (iravid) [inline ballista/rust/scheduler/src/scheduler_server/grpc.rs:413]: I think we need to check for a conflict here

Comment (fsnlla) [reply ballista/rust/scheduler/src/scheduler_server/grpc.rs]: I tweaked the `get_job_status` signature a bit, so now we can validate if this job_id exists

File: ballista/rust/core/src/serde/physical_plan/mod.rs [hunk 2]
```diff
@@ -831,7 +831,7 @@ impl AsExecutionPlan for PhysicalPlanNode {
                 .groups()
                 .iter()
                 .flatten()
-                .map(|b| *b)
+                .cloned()
```
Comment (thinkharderdev) [inline ballista/rust/core/src/serde/physical_plan/mod.rs:834]: This should be `copied` I think

**Review by thinkharderdev — APPROVED** [review]

**Pull request merged**